fictadvisor-web/src/components/pages/personal-teacher-page/personal-teacher-tabs/components/complaint-popup/ComplaintPopup.tsx

```diff
@@ -31,68 +32,68 @@ const ComplaintPopup: FC<ComplaintPopupProps> = ({
 
-  const handleSubmit = async (data: Complaint) => {
-    const { fullName, groupId, message, title } = data;
-    const complaint: Complaint = {
-      message,
-      title,
-      fullName: undefined,
-      groupId: undefined,
-    };
-    if (fullName) complaint.fullName = fullName;
-    if (groupId) complaint.groupId = groupId;
-
-    try {
-      await TeacherAPI.postTeacherComplaint(
-        teacherId,
-        convertEmptyStringToUndefined(complaint),
-      );
-      toast.success(
-        'Скарга надіслана',
-        'Ми успішно отримали скаргу на викладача та вже працюємо над нею',
-        5000,
-      );
-      setIsOpen(false);
-    } catch (error) {
-      displayError(error);
-    }
-  };
+  const handleSubmit = useCallback(
+    async (data: Complaint) => {
+      try {
+        await TeacherAPI.postTeacherComplaint(
+          teacherId,
+          convertEmptyStringToUndefined(data),
+        );
+        toast.success(
+          'Скарга надіслана',
+          'Ми успішно отримали скаргу на викладача та вже працюємо над нею',
+          5000,
+        );
+        setIsOpen(false);
+      } catch (error) {
+        displayError(error);
+      }
+    },
+    [teacherId, toast, displayError, setIsOpen],
+  );
 
   return (
     <Formik
-      enableReinitialize={true}
+      enableReinitialize
       validationSchema={validationSchema}
       initialValues={initialValues}
-      validateOnChange={true}
+      validateOnChange
       onSubmit={handleSubmit}
     >
-      {({}) => (
+      {({ handleSubmit, resetForm }) => (
         <Form>
           <Popup
             sx={styles.popup}
             open={isOpen}
             content={<ComplaintPopupContent />}
-            title={'Лишити скаргу на викладача'}
+            title="Лишити скаргу на викладача"
             firstButton={
-              <Box>
+              <Box sx={{ marginTop: '18px', marginBottom: '30px' }}>
                 <Button
                   text="Скасувати"
-                  size={ButtonSize.MEDIUM}
+                  size={isTablet ? ButtonSize.SMALL : ButtonSize.MEDIUM}
                   variant={ButtonVariant.OUTLINE}
-                  onClick={() => setIsOpen(false)}
+                  onClick={() => {
+                    resetForm();
+                    setIsOpen(false);
+                  }}
                 />
               </Box>
             }
-            hasCross={true}
-            onClose={() => setIsOpen(false)}
+            hasCross
+            onClose={() => {
+              resetForm();
+              setIsOpen(false);
+            }}
             secondButton={
-              <Box>
+              <Box sx={{ marginTop: '18px', marginBottom: '30px' }}>
                 <Button
-                  size={ButtonSize.MEDIUM}
+                  size={isTablet ? ButtonSize.SMALL : ButtonSize.MEDIUM}
                   variant={ButtonVariant.FILLED}
                   text="Лишити скаргу"
-                  type="submit"
```

Comment:
Is it now working with type="submit"? Strange

Reply:
it doesnt work with type="submit"